pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/SegmentPreProcessor.java

```diff
@@ -39,196 +37,177 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+
 /**
  * mmap()'s the segment and performs any pre-processing to generate inverted index
  * This can be slow
  */
 public class SegmentPreProcessor implements AutoCloseable {
-  private static Logger LOGGER = LoggerFactory.getLogger(SegmentPreProcessor.class);
+  private final static Logger LOGGER = LoggerFactory.getLogger(SegmentPreProcessor.class);
 
-  private final File indexDir;
-  private SegmentDirectory segmentDirectory;
-  private final SegmentMetadataImpl metadata;
-  private final Optional<IndexLoadingConfigMetadata> indexConfig;
-  //private IndexLoadingConfigMetadata indexConfig;
+  private final File _indexDir;
+  private final SegmentDirectory _segmentDirectory;
+  private final SegmentMetadataImpl _segmentMetadata;
+  private final String _segmentName;
+  private final SegmentVersion _segmentVersion;
+  private final IndexLoadingConfigMetadata _indexConfig;
 
-  SegmentPreProcessor(File indexDir, SegmentMetadataImpl metadata,
-      IndexLoadingConfigMetadata indexConfig) {
+  SegmentPreProcessor(File indexDir, SegmentMetadataImpl segmentMetadata, IndexLoadingConfigMetadata indexConfig) {
     Preconditions.checkNotNull(indexDir);
-    Preconditions.checkNotNull(metadata);
-    Preconditions.checkState(indexDir.exists(), "Segment directory: {} does not exist", indexDir);;
+    Preconditions.checkNotNull(segmentMetadata);
+    Preconditions.checkState(indexDir.exists(), "Segment directory: {} does not exist", indexDir);
     Preconditions.checkState(indexDir.isDirectory(), "Segment path: {} is not a directory", indexDir);
 
-    this.indexDir = indexDir;
-    this.metadata = metadata;
-    this.indexConfig = Optional.fromNullable(indexConfig);
-    // always use mmap. That's safest and performs well without impact from
-    // -Xmx params
-    // This is not final load of the segment
-    segmentDirectory = SegmentDirectory.createFromLocalFS(indexDir, metadata, ReadMode.mmap);
+    _indexDir = indexDir;
+    _segmentMetadata = segmentMetadata;
+    _segmentName = segmentMetadata.getName();
+    _segmentVersion = SegmentVersion.valueOf(_segmentMetadata.getVersion());
+    _indexConfig = indexConfig;
+    // Always use mmap to load the segment because it is safest and performs well without impact from -Xmx params.
+    // This is not the final load of the segment.
+    _segmentDirectory = SegmentDirectory.createFromLocalFS(indexDir, segmentMetadata, ReadMode.mmap);
   }
 
   public void process()
       throws IOException {
-    SegmentDirectory.Writer segmentWriter = null;
-    try {
-      segmentWriter = segmentDirectory.createWriter();
+    try (SegmentDirectory.Writer segmentWriter = _segmentDirectory.createWriter()) {
       addRemoveInvertedIndices(segmentWriter);
-    } finally {
```

Comment:
Yes, I know that, and I checked the save() method, which is a no-op. If we want to trigger saveAndClose() in finally block, I think we should overwrite close() to perform saveAndClose(). What do you think?